src/index.ts

```diff
@@ -1,28 +1,120 @@
+'use strict';
+
 import { PreCompiler } from "gherking";
-import { /* TODO */ } from "gherkin-ast";
+import { Scenario, ScenarioOutline, Examples, Tag, Rule, Element, Feature } from "gherkin-ast";
+import { FilterConfig } from "./types";
 // eslint-disable-next-line @typescript-eslint/no-var-requires
-const debug = require("debug")("gpc:template");
+const debug = require("debug")("gpc:filter");
+
+const checkDuplication = (one: Array<string>, other: Array<string>) => {
+    return one.filter((element: string) => other.includes(element));
+}
+
+const checkConfig = (config: FilterConfig) => {
+    let invalidScenarioTags: Array<string> = [];
+    let invalidRuleTags: Array<string> = []
+    if (config.includeScenarioTags && config.excludeScenarioTags) {
+        invalidScenarioTags = checkDuplication(config.includeScenarioTags, config.excludeScenarioTags)
+    }
+    if (config.includeRuleTags && config.excludeRuleTags) {
+        invalidRuleTags = checkDuplication(config.includeRuleTags, config.excludeRuleTags)
+    }
+    if (invalidScenarioTags.length || invalidRuleTags.length) {
+        throw new Error(`The following tag(s) are present as both inclusion and exclusion: ${invalidScenarioTags.concat(invalidRuleTags)}`);
+    }
+}
+
+class Filter implements PreCompiler {
```

Comment:
In general, you implemented the filtering logic at the event method level. I think the pre methods would be more suitable for this logic.